### PR TITLE
add release notes for 0.157.1-t

### PR DESCRIPTION
**Pull request by mattsfuller**
draft of release notes

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+* Additional Varchar(x) and Char(x) function implementations
+* Additional Decimal functions implementations
+
+Documentation
+-------------
+* Installation of Presto via Presto Admin
+* Presto Tuning Guide
+
+
+Performance
```
Comment (kbajda) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:24]: We should probably mention our work around Query Optimizer (link to that new section in the docs) that includes statistics for Hive connector, automatic join selection, cross join avoidance.

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+SQL
+---
+* Support correlated scalar aggregation subqueries
+* Support correlated scalar subqueries
+* Additional prepared statement syntax: DESCRIBE INPUT and DESCRIBE OUTPUT
+
+
+Data Types
+----------
+* Additional Varchar(x) and Char(x) function implementations
```
Comment (maciejgrzybek) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst]: How about "specialization for functions" or "Native functions for varchar(x) and char(x) types"? I believe before those patches we also supported varchar(x) and char(x) though through coercion.

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+* Additional Decimal functions implementations
```
Comment (maciejgrzybek) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst]: ditto

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  
+Bugs Fixed
+----------
+* Fix issue “Hive table is corrupt. It is declared as being bucketed, but the files do not match the bucketing declaration. The number of files in the directory (1) does not match the declared.” by fixing support for Hive bucketed tables. See option hive.multi-file-bucketing.enabled in the Presto Hive connector documentation.
+* Allow empty partitions for clustered hive tables
+* Fix issue “low must be less than or equal to high” that can occur with ORC and Character data.
+* Fix incorrect stream property derivations from GroupIdNode
+* Remove broken %w specifier for MySQL date functions
+* RPM to include installation of the Memory connector.
+
```
Comment (maciejgrzybek) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:75]: How about https://github.com/prestodb/presto/pull/6218 ?

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+* The installation method as documented on `prestodb.io <https://prestodb.io/docs/0.157/installation/deployment.html>`_.
+* Web Connector for Tableau
+* The following connectors:
+
+  * Accumulo Connector
+  * Kafka Connector
+  * Local File Connector
+  * MongoDB Connector
+  * Redis Connector
+
```
Comment (maciejgrzybek) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:93]: InMemory connector?

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 6]
```diff
@@ -0,0 +1,168 @@
+=================
+Release 0.157.1-t
+=================
+
+Presto 0.157.1-t is equivalent to Presto release 0.157.1, with some additional features and patches.
+
+Security
+--------
+* LDAP Authentication
```
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst]: We need to add something like "Support for `SHOW GRANTS` added for Hive connector". Either here, or preferably, in a separate section that lists features added for hive connector.

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst (hunk 5 above)
Comment (kbajda) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:93]: It's not supported for customers

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 7]
```diff
@@ -6,9 +6,23 @@ Presto 0.157.1-t is equivalent to Presto release 0.157.1, with some additional f
 
 Security
 --------
-* LDAP Authentication
+LDAP Authentication
+
```
Comment (anusudarsan) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst]: @mattsfuller Can you review this. You can squash this commit before merging

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+Bugs Fixed
+----------
+* Fix issue “Hive table is corrupt. It is declared as being bucketed, but the files do not match the bucketing declaration. The number of files in the directory (1) does not match the declared.” by fixing support for Hive bucketed tables. See option hive.multi-file-bucketing.enabled in the Presto Hive connector documentation.
+* Allow empty partitions for clustered hive tables
+* Fix issue “low must be less than or equal to high” that can occur with ORC and Character data.
+* Fix incorrect stream property derivations from GroupIdNode
+* Remove broken %w specifier for MySQL date functions
+* RPM to include installation of the Memory connector.
+
+
```
Comment (cawallin) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:78]: Add a separate section about the following bug fixes ported from 158:
* Fix query failures for connectors that produce non-remotely accessible splits.
* Fix non-linear performance issue when parsing certain SQL expressions.

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+* Optimize DictionaryBlock.copyPositions()
+  
+Bugs Fixed
+----------
+* Fix issue “Hive table is corrupt. It is declared as being bucketed, but the files do not match the bucketing declaration. The number of files in the directory (1) does not match the declared.” by fixing support for Hive bucketed tables. See option hive.multi-file-bucketing.enabled in the Presto Hive connector documentation.
+* Allow empty partitions for clustered hive tables
+* Fix issue “low must be less than or equal to high” that can occur with ORC and Character data.
+* Fix incorrect stream property derivations from GroupIdNode
+* Remove broken %w specifier for MySQL date functions
+* RPM to include installation of the Memory connector.
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:72]: Why is this a bug?

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+
```
Comment (rschlussel-zz) [inline presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:75]: Another bug fix: Fix wrong results with nested except statements

**Review by rschlussel-zz — COMMENTED** [review]
General question: aren't release notes supposed to build on each other about what things are different from release to release?  Why are there parts (the unsupported features and limitations sections) that we include in the release notes for every release.  Maybe that indicates that they warrant their own page of documentation.

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst (hunk 4 above)
Comment (mattsfuller) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:75]: isn't that already included in 0.157.1?

Comment (maciejgrzybek) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:75]: You're right.

File: presto-docs/src/main/sphinx/release/release-0.157.1-t.rst (hunk 9 above)
Comment (maciejgrzybek) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:72]: Why this is here if we don't support InMemory for customers (https://github.com/Teradata/presto/pull/460#discussion_r91100614)?

Comment (kbajda) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:72]: InMemory connector is for experimental purposes only. But it's OK to experiment.

Comment (maciejgrzybek) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:72]: So why don't we include it in added features with annotation it's experimental then? ;)

Comment (kbajda) [reply presto-docs/src/main/sphinx/release/release-0.157.1-t.rst:72]: Oh, I will let @mattsfuller to handle this but I suggest putting "InMemory" connector documented under "community-supported" section, not "Teradata-supported". In the description of that connector (http://teradata.github.io/presto/docs/current/connector/memory.html) we clearly say: "This connector is in early experimental stage it is not recommended to use it in a production environment. The intended use of this connector at this point is purely for testing."

**Pull request closed without merging**